src/quad_expr.jl

```diff
@@ -198,13 +198,20 @@ function Base.convert(::Type{GenericQuadExpr{C, V}}, v::Union{Real,AbstractVaria
 end
 GenericQuadExpr{C, V}() where {C, V} = zero(GenericQuadExpr{C, V})
 
+"""
+    moi_quadratic_term(t::Tuple)
+
+Return the MOI.ScalarQuadraticTerm for the quadratic term `t`, element of the
+[`quadterms`](@ref) iterator.
+"""
+function moi_quadratic_term(t::Tuple)
+    return MOI.ScalarQuadraticTerm(t[2] == t[3] ? 2t[1] : t[1], index(t[2]),
+                                   index(t[3]))
+end
 function MOI.ScalarQuadraticFunction(q::QuadExpr)
     assert_isfinite(q)
-    qterms = MOI.ScalarQuadraticTerm{Float64}[MOI.ScalarQuadraticTerm(
-                                               t[2] == t[3] ? 2t[1] : t[1],
```

Comment:
Was this logic for the off-diagonals wrong? Why don't the tests change for this?

Reply:
It was not wrong I just refactored it into a separate function but its equivalent

Reply:
Got it.